app/controllers/find/search/locations_controller.rb

```diff
@@ -11,6 +11,12 @@ def start; end
       def new; end
 
       def create
+        # if searching by specific provider we have to filter params
+        if provider_option_selected?
+          redirect_to next_step(params)
+          return
```

Comment:
no need for this return?

Reply:
Render and/or redirect were called multiple times in this action. Please note that you may only call render OR redirect, and at most once per action. Also note that neither redirect nor render terminate execution of the action, so if you want to exit an action after redirecting, you need to do something like "redirect_to(...) and return".